adafruit_ht16k33/segments.py

```diff
@@ -211,21 +215,62 @@ def _text(self, text):
         for character in text:
             self._push(character)
 
-    def _number(self, number):
-        """Display the specified decimal number."""
+    '''
+		Display a floating point or integer number on the Adafruit HT16K33 based displays
+		
+		Param: number - The floating point or integer number to be displayed, which must be
+			in the range 0 (zero) to 9999 for integers and floating point or integer numbers
+			and between 0.0 and 999.0 or 99.00 or 9.000 for floating point numbers.
+		Param: decimal - The number of decimal places for a floating point number if decimal
+			is greater than zero, or the input number is an integer if decimal is zero.
+
+        Returns: The output text string to be displayed.'''
+    def _number(self, number, decimal=0):
         auto_write = self._auto_write
         self._auto_write = False
-        string = "{}".format(number)
-        if len(string) > 4:
-            if string.find('.') > 4:
-                raise ValueError("Overflow")
-        self.fill(False)
-        places = 4
-        if '.' in string:
-            places += 1
-        self._text(string[:places])
+        stnum = str(number)
+        dot = stnum.find('.')
+
+        if ((len(stnum) > 5) or ((len(stnum) > 4) and (dot < 0))):
+            raise ValueError("Input overflow - {0} is too large for the display!".format(number))
+
+        if dot < 0:
+			#	No decimal point (Integer)
+            places = len(stnum)
+        else:
+            places = len(stnum[:dot])
+
+        if self.debug:
+            print("(1) number = {0}, places = {1}, decimal = {2}, dot = {3}, stnum = '{4}'".format(number, places, decimal, dot, stnum))
+
+        if ((places <= 0) and (decimal > 0)):
```

Comment:
This is failing pylint because of the parenthesis and `variable > 0` can be shortened to `variable`. I'd suggest changing this line to:
`if places <= 0 and decimal:`

Reply:
Same comment on zero checks.